src/selectors/clusterSelectors.js

```diff
@@ -144,36 +145,44 @@ export const selectTargetRelease = (state, cluster) => {
   if (!cluster || Object.keys(state.entities.releases.items).length === 0)
     return null;
 
-  const releases = state.entities.releases.items;
-  let availableVersions = Object.keys(releases).sort(cmp);
-
-  if (!state.main.loggedInUser.isAdmin) {
-    availableVersions = availableVersions.filter(
-      (release) => releases[release].active
-    );
-  }
+  const releases = Object.assign({}, state.entities.releases.items);
```

Comment:
The `selectTargetRelease()` function logic has been reworked:
-> It will return the next active version for all user types.
-> If there is no active version and the user is not an admin, return `null`.
-> If there is no active version and the user is an admin, return the next `inactive` version.
-> If there is no next `inactive` version and the user is an admin, return `null`

Reply:
Before this, was it possoble for Giant Swarm staff to upgrade cluster through incompatible versions (9->11 on AWS)? 

For a `9.0.9` cluster this implementation would allow me to upgrade to `11.0.1` and above.

Maybe we add a little heads up to the modal text which says something along `Please don't upgrade from 9 to 11 on AWS)?

Reply:
The API should prevent it,... and I thought we had some logic in Happa that also avoid offering an upgrade past that boundary (for all users), but maybe I am remembering it wrong.

Reply:
Upgrading from 9.x.x to any newer major is blocked on the API side, so not a big deal.

Please note that for customers the `Upgrade` button will not be shown at all if the newer version has a major larger than `9`.

Here's the logic for showing the `Upgrade` button
https://github.com/giantswarm/happa/blob/0bb388ee730f8a8813e7c6044a694a09d3be9fff/src/selectors/clusterSelectors.js#L186

At the end there's also this check, that doesn't allow upgrading from V4 to V5 on AWS
https://github.com/giantswarm/happa/blob/0bb388ee730f8a8813e7c6044a694a09d3be9fff/src/utils/clusterUtils.js#L10